lib/Queue.rb

```diff
@@ -0,0 +1,38 @@
+
+class Queue
+  attr_reader  :data
+
+  def initialize (data = [])
+    @data = data
+    @e_cursor = 0
+    @d_cursor = 0
```

Comment:
I'm guessing these are enqueue_cursor, and dequeue_cursor, right? Could be clearer.